addons/website_sale/static/src/js/website_sale_delivery.js

```diff
@@ -365,7 +367,22 @@ publicWidget.registry.websiteSaleDelivery = publicWidget.Widget.extend({
         if (radio.checked && !this._shouldDisplayPickupLocations(ev) && !this.forceClickCarrier) {
             return;
         }
-        this.forceClickCarrier = false;
+
+        const pickupLocs = document.querySelectorAll('.o_show_pickup_locations');
+        pickupLocs?.forEach(loc => {
+            const parent = loc.parentElement;
+
+            const nameElement = parent.querySelector('.o_order_location_name');
+            const addressElement = parent.querySelector('.o_order_location_address');
+            if (nameElement) nameElement.textContent = '';
+            if (addressElement) addressElement.textContent = '';
+            parent.querySelector('.o_remove_order_location')?.remove();
+
+            const divDNone = parent.querySelector('.o_order_location').parentElement;
+            if (divDNone && !divDNone.classList.contains('d-none')) {
+                divDNone.classList.add('d-none');
+            }
+        });
```

Comment:
We can simplify it a bit to search directly for `o_order_location`. Do not forget that we do not want to do that when `this.forceClickCarrier` is true to not reset location on reload. Could you also add a comment?
```suggestion
        const orderLocs = document.querySelectorAll('.o_order_location');
        orderLocs.forEach(loc => {
                 loc.querySelector('.o_order_location_name').textContent = '';
                loc.querySelector('.o_order_location_address').textContent = '';
                const divDNone = loc.parentElement;
            if (!divDNone.classList.contains('d-none')) {
                divDNone.classList.add('d-none');
            }
        });
```